LayoutTestTests/LayoutTestCaseTests/LYTLayoutFailingTestSnapshotRecorderTests.m

```diff
@@ -0,0 +1,304 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+
+#import <XCTest/XCTest.h>
+#import "LYTLayoutFailingTestSnapshotRecorder.h"
+#import "LYTConfig.h"
+
+@interface LYTLayoutFailingTestSnapshotRecorderTests : XCTestCase
+
+@property (nonatomic) LYTLayoutFailingTestSnapshotRecorder *recorder;
+
+@end
+
+@implementation LYTLayoutFailingTestSnapshotRecorderTests
+
+- (void)setUp {
+    [super setUp];
+    self.recorder = [[LYTLayoutFailingTestSnapshotRecorder alloc] init];
+}
+
+- (void)tearDown {
+    self.recorder = nil;
+    [super tearDown];
+}
+
+- (void)testStartNewLogDeletesExisitingClassSnapshotDirectory {
+    //Setup directory and file to make sure file is deleted when we start a new log.
+    NSString *currentDirectory = [[NSBundle bundleForClass:self.class] bundlePath];
+    NSFileManager *fileManager = [NSFileManager defaultManager];
+    NSString *classDirectory = [currentDirectory stringByAppendingPathComponent:@"LayoutTestImages/LYTLayoutFailingTestSnapshotRecorderTests"];
+    NSString *testFilePath = [classDirectory stringByAppendingPathComponent:@"testFile.html"];
+    [fileManager createDirectoryAtPath:classDirectory withIntermediateDirectories:YES attributes:nil error:nil];
+    [fileManager createFileAtPath:testFilePath contents:nil attributes:nil];
+
+    [self.recorder startNewLogForClass:self.class];
+    
+    XCTAssertFalse([fileManager fileExistsAtPath:testFilePath]);
+}
+
+- (void)testStartNewLogCreatesIndexHTMLFileWithExpectedContent {
+    [self.recorder startNewLogForClass:self.class];
+    
+    NSString *actualHTML = [self indexHTMLFile];
+    XCTAssertEqualObjects([self expectedStartOfFileHTML], actualHTML);
+}
+
+- (void)testFinishLogAddsExpectedHTMLToIndexFile {
+    [self.recorder startNewLogForClass:self.class];
+    [self.recorder finishLog];
+    
+    NSString *actualHTML = [self indexHTMLFile];
+    NSString *expectedHTML = [NSString stringWithFormat:@"%@%@", [self expectedStartOfFileHTML], [self expectedEndOfFileHTML]];
+    XCTAssertEqualObjects(expectedHTML, actualHTML);
+}
+
+- (void)testSaveImageOfCurrentViewCreatesDirectoryFromInvocationSelectorName {
+    [self.recorder startNewLogForClass:self.class];
+    
+    [self.recorder saveImageOfView:nil withData:nil fromInvocation:self.invocation failureDescription:@""];
+    
+    NSString *directoryPath = [self pathForCurrentInvocation];
+    BOOL isDirectory = NO;
+    BOOL fileExisits = [[NSFileManager defaultManager] fileExistsAtPath:directoryPath isDirectory:&isDirectory];
+    
+    XCTAssertTrue(fileExisits);
+    XCTAssertTrue(isDirectory);
+}
+
+- (void)testSaveImageOfCurrentViewSavesImageToDisk {
+    UIView *testView = [[UIView alloc] initWithFrame:(CGRect){0, 0, 100, 100}];
+    [self.recorder startNewLogForClass:self.class];
+    
+    [self.recorder saveImageOfView:testView withData:@{@"key" : @"value"} fromInvocation:self.invocation failureDescription:@""];
+    
+    NSString *dataHash = @"";
+#ifdef __LP64__
```

Comment:
Instead of this, why not just call .description.hash on the data? That way it'll be future proof too?